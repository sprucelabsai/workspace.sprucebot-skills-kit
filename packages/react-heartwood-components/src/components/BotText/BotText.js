// @flow
import React from 'react'
import type { Node } from 'react'
import cx from 'classnames'

type Props = {
	/** Use for simple text. */
	text?: string,

	/** May be used instead of text, i.e. if the Bot Text needs to include a link. */
	children?: Node,

	/** Optional class to add. */
	className?: string
}

const BotText = (props: Props) => {
	const { text, children, className, ...rest } = props

	if (!children && !text) {
		// TODO: Handle logging
		// console.error('Bot Text must either have children or text')
		return null
	}

	if (text && !children) {
		return (
			<p
				className={cx('bot-text', className)}
				dangerouslySetInnerHTML={{ __html: text }}
				{...rest}
			/>
		)
	}
	if (children && !text) {
		return (
			<p className={cx('bot-text', className)} {...rest}>
				{children}
			</p>
		)
	}
	return null
}

export default BotText
