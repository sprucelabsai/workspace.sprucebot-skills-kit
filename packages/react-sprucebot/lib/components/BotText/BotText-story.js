"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _react = _interopRequireDefault(require("react"));

var _react2 = require("@storybook/react");

var _react3 = require("@storybook/addon-knobs/react");

var _Container = _interopRequireDefault(require("../Layout/Container/Container"));

var _BotText = _interopRequireDefault(require("./BotText"));

var stories = (0, _react2.storiesOf)('Bot Text', module);
stories.addDecorator(_react3.withKnobs);
stories.add('Bot Text', function () {
  return _react.default.createElement(_Container.default, {
    size: "small"
  }, _react.default.createElement(_BotText.default, {
    text: (0, _react3.text)('text', 'Hey there from Sprucebot ðŸŒ²ðŸ¤–'),
    className: (0, _react3.text)('className', '')
  }));
});