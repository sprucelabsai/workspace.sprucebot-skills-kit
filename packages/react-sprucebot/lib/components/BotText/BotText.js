'use strict';

Object.defineProperty(exports, "__esModule", {
	value: true
});

var _styledComponents = require('styled-components');

var _styledComponents2 = _interopRequireDefault(_styledComponents);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var BotText = _styledComponents2.default.div.attrs({
	className: 'bot__text'
}).withConfig({
	displayName: 'BotText',
	componentId: 'udhfms-0'
})(['word-wrap:break-word;overflow-wrap:break-word;word-break:break-word;']);

exports.default = BotText;