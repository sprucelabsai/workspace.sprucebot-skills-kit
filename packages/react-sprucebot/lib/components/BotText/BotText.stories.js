'use strict';

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _react3 = require('@storybook/react');

var _addonActions = require('@storybook/addon-actions');

var _react4 = require('@storybook/addon-knobs/react');

var _storybookReadme = require('storybook-readme');

var _addonInfo = require('@storybook/addon-info');

var _BotText = require('./BotText');

var _BotText2 = _interopRequireDefault(_BotText);

var _BotText3 = require('./BotText.md');

var _BotText4 = _interopRequireDefault(_BotText3);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var stories = (0, _react3.storiesOf)('BotText', module);

stories.addDecorator(_react4.withKnobs);

stories.add('Interactive', (0, _storybookReadme.withReadme)(_BotText4.default, (0, _addonInfo.withInfo)()(function () {
	return _react2.default.createElement(
		_BotText2.default,
		null,
		(0, _react4.text)('BotText', 'Hey there from Sprucebot ðŸŒ²ðŸ¤–')
	);
})));