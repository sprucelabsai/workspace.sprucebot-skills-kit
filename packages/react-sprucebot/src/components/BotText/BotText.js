import styled from 'styled-components'

const BotText = styled.div.attrs({
	className: 'bot__text'
})`
	word-wrap: break-word;
	overflow-wrap: break-word;
	word-break: break-word;
`

export default BotText
