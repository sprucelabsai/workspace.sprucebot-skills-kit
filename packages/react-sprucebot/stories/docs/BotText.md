# BotText

I'm some bot text. I'm great for quick hints, tips, shout outs, etc.
This text comes right from Sprucebot, so make sure it's on brand.
