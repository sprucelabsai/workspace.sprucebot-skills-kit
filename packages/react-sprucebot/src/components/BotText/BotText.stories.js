import React from 'react'
import { storiesOf } from '@storybook/react'
import { action } from '@storybook/addon-actions'
import { withKnobs, text, boolean, number } from '@storybook/addon-knobs/react'
import { withReadme, withDocs } from 'storybook-readme'
import { withInfo } from '@storybook/addon-info'
import BotText from './BotText'
import readme from './BotText.md'

const stories = storiesOf('BotText', module)

stories.addDecorator(withKnobs)

stories.add(
	'Interactive',
	withReadme(
		readme,
		withInfo()(() => (
			<BotText>{text('BotText', 'Hey there from Sprucebot ğŸŒ²ğŸ¤–')}</BotText>
		))
	)
)
