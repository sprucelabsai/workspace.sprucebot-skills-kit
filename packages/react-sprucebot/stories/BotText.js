import React from 'react'
import { storiesOf } from '@storybook/react'
import { action } from '@storybook/addon-actions'
import { withKnobs, text, boolean, number } from '@storybook/addon-knobs/react'
import { withReadme, withDocs } from 'storybook-readme'
import { withInfo } from '@storybook/addon-info'
import BotText from '../lib/components/BotText/BotText'
import readme from './docs/BotText.md'

const stories = storiesOf('BotText', module)

stories.addDecorator(withKnobs)

stories.add(
	'Interactive',
	withReadme(
		readme,
		withInfo()(() => (
			<BotText>{text('BotText', 'Hey there from Sprucebot ğŸŒ²ğŸ¤–')}</BotText>
		))
	)
)
