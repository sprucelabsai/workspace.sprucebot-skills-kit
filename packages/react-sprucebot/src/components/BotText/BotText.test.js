import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'

import BotText from './BotText'

test('it renders', () => {
	const tree = renderer.create(<BotText>Bottext children</BotText>).toJSON()
	expect(tree).toMatchSnapshot()
})
