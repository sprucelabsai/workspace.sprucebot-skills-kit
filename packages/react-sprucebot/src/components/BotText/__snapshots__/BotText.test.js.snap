// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders 1`] = `
.c0 {
  word-wrap: break-word;
  overflow-wrap: break-word;
  word-break: break-word;
}

<div
  className="bot__text c0"
>
  Bottext children
</div>
`;
